presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -1755,6 +1763,67 @@ private static Number toNumber(Object value)
         return (maxRows > 0) ? new LengthLimitedIterator<>(rowsIterator, maxRows) : rowsIterator;
     }
 
+    private static class AsyncIterator<T>
+            implements Iterator<T>
+    {
+        private static final int MAX_QUEUED_ROWS = 50_000;
+        private static ThreadFactoryBuilder threadFactoryBuilder = new ThreadFactoryBuilder().setDaemon(true)
+                .setNameFormat("async-client-datafetch-%d");
+
+        private static final ExecutorService executorService = Executors.newCachedThreadPool(threadFactoryBuilder.build());
```

Comment:
Static import `newCachedThreadPool` and inline the call to the thread factory. Also, let's name the thread pool so that it's clear it is for Presto JDBC:
```java
private static final ExecutorService executorService = newCachedThreadPool(daemonThreadsNamed("Presto JDBC worker-%d"));